modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionValueResourceTest.java

```diff
@@ -6,15 +6,331 @@
+	protected ProductOptionValue randomPatchProductOptionValue()
+		throws Exception {
+
+		return randomProductOptionValue();
+	}
+
+	@Override
+	protected ProductOptionValue randomProductOptionValue() throws Exception {
+		return new ProductOptionValue() {
+			{
+				id = RandomTestUtil.randomLong();
+				key = StringUtil.toLowerCase(RandomTestUtil.randomString());
+				name = LanguageUtils.getLanguageIdMap(
+					RandomTestUtil.randomLocaleStringMap());
+				preselected = RandomTestUtil.randomBoolean();
+				priority = RandomTestUtil.randomDouble();
+				quantity = BigDecimal.valueOf(RandomTestUtil.randomDouble());
+				skuExternalReferenceCode =
+					_cpInstance.getExternalReferenceCode();
+				skuId = _cpInstance.getCPInstanceId();
+				unitOfMeasureKey = StringUtil.toLowerCase(
+					RandomTestUtil.randomString());
+			}
+		};
+	}
+
+	@Ignore
+	@Override
+	@Test
+	protected ProductOptionValue
+			testDeleteProductOptionValue_addProductOptionValue()
+		throws Exception {
+
+		return super.testDeleteProductOptionValue_addProductOptionValue();
+	}
+
+	@Override
+	protected ProductOptionValue
+			testPatchProductOptionValue_addProductOptionValue()
+		throws Exception {
+
+		ProductOptionValue randomProductOptionValue =
+			randomProductOptionValue();
+
+		return productOptionValueResource.postProductOptionIdProductOptionValue(
+			_cpDefinitionOptionRel.getCPDefinitionOptionRelId(),
+			randomProductOptionValue);
+	}
+
+	@Override
+	protected ProductOptionValue
+			testPostProductOptionIdProductOptionValue_addProductOptionValue(
+				ProductOptionValue productOptionValue)
+		throws Exception {
+
+		return productOptionValueResource.postProductOptionIdProductOptionValue(
+			_cpDefinitionOptionRel.getCPDefinitionOptionRelId(),
+			productOptionValue);
+	}
+
+	private void _testPatchProductOptionIdProductOptionValueWithSkuExternalReferenceCode()
+		throws Exception {
+
+		ProductOptionValue postProductOptionValue =
+			testPatchProductOptionValue_addProductOptionValue();
+
+		ProductOptionValue randomPatchProductOptionValue =
+			randomProductOptionValue();
+
+		CPInstance cpInstance = CPTestUtil.addCPInstanceWithRandomSku(
+			testGroup.getGroupId(), BigDecimal.TEN);
+
+		randomPatchProductOptionValue.setSkuId(0L);
+
+		randomPatchProductOptionValue.setSkuExternalReferenceCode(
+			cpInstance.getExternalReferenceCode());
+
+		ProductOptionValue patchProductOptionValue =
+			productOptionValueResource.patchProductOptionValue(
+				postProductOptionValue.getId(), randomPatchProductOptionValue);
+
+		ProductOptionValue expectedPatchProductOptionValue =
+			postProductOptionValue.clone();
+
+		BeanTestUtil.copyProperties(
+			randomPatchProductOptionValue, expectedPatchProductOptionValue);
+
+		ProductOptionValue getProductOptionValue =
+			productOptionValueResource.getProductOptionValue(
+				patchProductOptionValue.getId());
+
+		assertEquals(expectedPatchProductOptionValue, getProductOptionValue);
+		assertValid(getProductOptionValue);
+	}
```

Comment:
please add the check if the two ids are the same and the two ercs are the same

Reply:
like you did in _testPostProductOptionIdProductOptionValueWithSkuExternalReferenceCode